File: CS/GridControlTypeConverter/LocalizedEnumConverter/LocalizedEnumConverter.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Columns;
using System.Collections;
using System.Globalization;

namespace GridControlTypeConverter
{
    /// <summary>
    /// Defines a type converter for enum types defined in this project
    /// </summary>
    class LocalizedEnumConverter : ResourceEnumConverter
    {
        /// <summary>
        /// Create a new instance of the converter using translations from the given resource manager
        /// </summary>
        /// <param name="type"></param>
        public LocalizedEnumConverter(Type type)
            : base(type, Properties.Resources.ResourceManager)
        {

        }
        public override bool CanConvertFrom(ITypeDescriptorContext context, Type sourceType)
        {
            if(sourceType == typeof(MyObject.MyEnum))
                return true;
            return base.CanConvertFrom(context, sourceType);
        }
    }
}

File: VB/GridControlTypeConverter/LocalizedEnumConverter/LocalizedEnumConverter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Columns
Imports System.Collections
Imports System.Globalization

Namespace GridControlTypeConverter
	''' <summary>
	''' Defines a type converter for enum types defined in this project
	''' </summary>
	Friend Class LocalizedEnumConverter
		Inherits ResourceEnumConverter
		''' <summary>
		''' Create a new instance of the converter using translations from the given resource manager
		''' </summary>
		''' <param name="type"></param>
		Public Sub New(ByVal type As Type)
			MyBase.New(type, My.Resources.ResourceManager)

		End Sub
		Public Overrides Function CanConvertFrom(ByVal context As ITypeDescriptorContext, ByVal sourceType As Type) As Boolean
			If sourceType Is GetType(MyObject.MyEnum) Then
				Return True
			End If
			Return MyBase.CanConvertFrom(context, sourceType)
		End Function
	End Class
End Namespace
